packages/terra-tag/src/Tag.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './Tag.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  TAB: 9,
+};
+
+const tagTheme = {
+  NORMAL: 'normal',
+};
+
+const propTypes = {
+  /**
+   * Sets the href. When set will render the component as an anchor tag.
+   */
+  href: PropTypes.string,
+  /**
+   * An optional icon.
+   */
+  icon: PropTypes.element,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Callback function triggered when tag loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Callback function triggered when tag gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
+   */
+  onKeyUp: PropTypes.func,
+  /**
+   * Sets the tag text.
+   */
+  text: PropTypes.string.isRequired,
+  /**
+   * Sets the tag theme.
+   */
+  theme: PropTypes.string,
```

Comment:
What is the need for the theme prop? What are the available themes a users can pass to this prop?

Reply:
Currently only normal theme is available , but in future scope we have an Inverse theme. So I added the theme prop.